examples/self-diagnostics/src/main.rs

```diff
@@ -0,0 +1,150 @@
+    // - Restrict `hyper`, `tonic`, and `reqwest` to `error` level logs only.
+    // This ensures events generated from these crates within the OTLP Exporter are not looped back,
+    // thus preventing infinite event generation.
+    // Note: This will also drop events from these crates used outside the OTLP Exporter.
+    // For more details, see: https://github.com/open-telemetry/opentelemetry-rust/issues/761
+    let filter = EnvFilter::new("info")
+        .add_directive("hyper=error".parse().unwrap())
+        .add_directive("tonic=error".parse().unwrap())
+        .add_directive("reqwest=error".parse().unwrap());
+    let cloned_provider = provider.clone();
+    let layer = layer::OpenTelemetryTracingBridge::new(&cloned_provider);
+    tracing_subscriber::registry()
+        .with(filter)
+        .with(layer)
+        .init();
+    provider
+}
+
+fn init_meter_provider() -> opentelemetry_sdk::metrics::SdkMeterProvider {
+    let provider = opentelemetry_otlp::new_pipeline()
+        .metrics(opentelemetry_sdk::runtime::Tokio)
+        .with_period(std::time::Duration::from_secs(1))
+        .with_exporter(
+            opentelemetry_otlp::new_exporter()
+                .http()
+                .with_endpoint("http://localhost:4318/v1/metrics"),
+        )
+        .build()
+        .unwrap();
+    let cloned_provider = provider.clone();
+    global::set_meter_provider(cloned_provider);
+    provider
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
+    // Set the custom error handler
+    if let Err(err) = set_error_handler(custom_error_handler) {
+        eprintln!("Failed to set custom error handler: {}", err);
+    }
+
+    let logger_provider = init_logger_provider();
+
+    // Initialize the MeterProvider with the stdout Exporter.
+    let meter_provider = init_meter_provider();
+
+    // Create a meter from the above MeterProvider.
+    let meter = global::meter("example");
+    // Create a Counter Instrument.
+    let counter = meter.u64_counter("my_counter").init();
+
+    // Record measurements with unique key-value pairs to exceed the cardinality limit
+    // and trigger error message
+    for i in 0..3000 {
+        counter.add(
+            10,
+            &[KeyValue::new(
+                format!("mykey{}", i),
+                format!("myvalue{}", i),
+            )],
+        );
+    }
+
+    let (tx, rx) = channel();
+
+    ctrlc::set_handler(move || tx.send(()).expect("Could not send signal on channel."))
+        .expect("Error setting Ctrl-C handler");
+
+    println!("Press Ctrl-C to continue...");
+    rx.recv().expect("Could not receive from channel.");
+    println!("Got Ctrl-C, Doing shutdown and existing.");
+
+    // Metrics are exported by default every 30 seconds when using stdout exporter,
+    // however shutting down the MeterProvider here instantly flushes
+    // the metrics, instead of waiting for the 30 sec interval.
+    meter_provider.shutdown()?;
```

Comment:
Use `force_flush` instead of `shutdown`?

Reply:
This was delibrate to call shutdown twice, and trigger "already shutdown" error.

Reply:
The collector logs added in the README file do not show anything for "already shutdown" error though. There is only any entry for exceeding cardinality limit.

Reply:
Yes, I meant the extra shutdown was to demonstrate that the error is triggered but not logged, as the pipeline is already closed with the first shutdown. But it seems it was causing more confusion, so have removed the second shutdown :)